pycon/tests/test_export.py

```diff
@@ -0,0 +1,135 @@
+from pycon.sponsorship.models import SponsorLevel, Sponsor, Benefit
+from pycon.tests.factories import PyConTutorialProposalFactory
+from symposion.conference.models import Conference, current_conference, Section
+from symposion.proposals.models import ProposalKind
+from symposion.proposals.tests.factories import ProposalBaseFactory
+from symposion.schedule.models import Schedule, Slot, Day, SlotKind
+from symposion.schedule.tests.factories import PresentationFactory
+
+
+class ProgramExportTest(TestCase):
+    fixtures = [
+        'fixtures/conference.json',
+        'fixtures/proposal_base.json'
+    ]
+
+    def setUp(self):
+        self.url = reverse('program_export')
+        Conference.objects.get_or_create(id=settings.CONFERENCE_ID)
+        conference = current_conference()
+        self.sponsor_level = SponsorLevel.objects.create(
+            conference=conference, name="Lead", cost=1)
+        self.sponsor = Sponsor.objects.create(
+            name="Big Daddy",
+            level=self.sponsor_level,
+            active=True,
+        )
+
+        # Create our benefits, of various types
+        self.text_benefit = Benefit.objects.create(name="text", type="text")
+        self.file_benefit = Benefit.objects.create(name="file", type="file")
+        # These names must be spelled exactly this way:
+        self.weblogo_benefit = Benefit.objects.create(
+            name="Web logo", type="weblogo")
+        self.printlogo_benefit = Benefit.objects.create(
+            name="Print logo", type="file")
+        self.advertisement_benefit = Benefit.objects.create(
+            name="Advertisement", type="file")
+
+        yesterday = now() - timedelta(hours=24)
+        tomorrow = now() + timedelta(hours=24)
+
+        for kind_name in ['Talk', 'Tutorial']:
+            slug = kind_name.lower()
+            section, __ = Section.objects.get_or_create(
+                conference=conference,
+                name__iexact=kind_name + "s",  # Section names are plural
+                defaults=dict(name=kind_name + "s",
+                              start_date=yesterday,
+                              end_date=tomorrow,
+                              slug=slug))
+            ProposalKind.objects.get_or_create(name__iexact=kind_name, defaults=dict(slug=slug, section=section))
+            Schedule.objects.get_or_create(section=section)
+
+    def test_talks_schedule(self):
+        section=Section.objects.get(name='Talks')
+        schedule=Schedule.objects.get(section=section)
+
+        day=Day.objects.create(schedule=schedule, date=date.today())
+        kind=SlotKind.objects.create(schedule=schedule, label="Foo")
+        slot = Slot.objects.create(
+            day=day,
+            kind=kind,
+            start=now().time(),
+            end=now().time(),
+        )
+        pres = PresentationFactory(
+            title="Faffa the dog",
+            abstract="Tinkerbell",
+            description="Not very much, thank you",
```

Comment:
Looks like many of these things are defined on the Factory, and not explicitly used here. Do these fields need to be set?